src/executor/update.rs

```diff
@@ -78,17 +80,45 @@ where
     Updater::new(query).exec(db).await
 }
 
-async fn exec_update_and_return_original<'a, A, C>(
-    query: UpdateStatement,
+async fn exec_update_and_return_updated<'a, A, C>(
+    mut query: UpdateStatement,
     model: A,
     db: &'a C,
 ) -> Result<A, DbErr>
 where
+    <A::Entity as EntityTrait>::Model: IntoActiveModel<A>,
     A: ActiveModelTrait,
     C: ConnectionTrait<'a>,
 {
-    Updater::new(query).check_record_exists().exec(db).await?;
-    Ok(model)
+    let db_backend = db.get_database_backend();
+    let found = match db_backend {
+        DbBackend::Postgres => {
+            query.returning(Returning::Columns(
+                <A::Entity as EntityTrait>::Column::iter()
+                    .map(|c| c.into_column_ref())
+                    .collect(),
+            ));
+            SelectorRaw::<SelectModel<<A::Entity as EntityTrait>::Model>>::from_statement(
+                db_backend.build(&query),
+            )
+            .one(db)
+            .await?
+        }
```

Comment:
But now, it seems impossible to produce the same error using returning.

Reply:
And that's why we have this error [here](https://github.com/SeaQL/sea-orm/runs/4118287744?check_suite_focus=true#step:6:201)
```
thread 'main' panicked at 'assertion failed: `(left == right)`

Diff < left / right > :
 Err(
<    Exec(
<        "Failed to find inserted item",
>    RecordNotFound(
>        "None of the database rows are affected",
     ),
 )

', tests/byte_primary_key_tests.rs:48:5
```

Reply:
Resolved on 6238736